pkg/maven/maven.go

```diff
@@ -89,31 +115,19 @@ func evaluateStdOut(config *ExecuteOptions) (*bytes.Buffer, io.Writer) {
 	return stdOutBuf, stdOut
 }
 
-func downloadSettingsIfURL(settingsFileOption, settingsFile string, client http.Downloader) (string, error) {
-	result := settingsFileOption
-	if strings.HasPrefix(settingsFileOption, "http:") || strings.HasPrefix(settingsFileOption, "https:") {
-		err := downloadSettingsFromURL(settingsFileOption, settingsFile, client)
-		if err != nil {
-			return "", err
-		}
-		result = settingsFile
-	}
-	return result, nil
-}
-
-func getParametersFromOptions(options *ExecuteOptions, client http.Downloader) ([]string, error) {
+func getParametersFromOptions(options *ExecuteOptions, utils mavenUtils) ([]string, error) {
 	var parameters []string
 
 	if options.GlobalSettingsFile != "" {
-		globalSettingsFileName, err := downloadSettingsIfURL(options.GlobalSettingsFile, "globalSettings.xml", client)
+		globalSettingsFileName, err := downloadSettingsIfURL(options.GlobalSettingsFile, ".pipeline/mavenGlobalSettings.xml", utils)
 		if err != nil {
 			return nil, err
 		}
 		parameters = append(parameters, "--global-settings", globalSettingsFileName)
 	}
 
 	if options.ProjectSettingsFile != "" {
-		projectSettingsFileName, err := downloadSettingsIfURL(options.ProjectSettingsFile, "projectSettings.xml", client)
+		projectSettingsFileName, err := downloadSettingsIfURL(options.ProjectSettingsFile, ".pipeline/mavenProjectSettings.xml", utils)
```

Comment:
Where is this new path used?

Reply:
It is returned by `downloadSettingsIfURL()`, so now contained in `projectSettingsFileName` and used from there. (`projectSettingsFileName` will either contain `options.ProjectSettingsFile` or `.pipeline/mavenProjectSettings.xml` in case the options pointed to a remote location.)